src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -82,11 +82,11 @@ public NuGetProject DefaultNuGetProject
 
         public event EventHandler<ActionsExecutedEventArgs> ActionsExecuted;
 
-        public VSSolutionManager()
+        public VSSolutionManager([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider)
```

Comment:
I like the idea of passing `IServiceProvider` to the constructor via MEF import.